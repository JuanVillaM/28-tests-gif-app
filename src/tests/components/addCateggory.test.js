import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { AddCategory } from '../../components/AddCategory';

describe('Tests in <addCategory />', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={ setCategories }/>);

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories }/>);
    });

    test('Should show himself correctly.', () => {
        expect( wrapper ).toMatchSnapshot();        
    });

    test('Should change the input.', () => {
        const input = wrapper.find('input');
        const value = 'Hello Juan';

        input.simulate('change', { target: { value } });

        expect( wrapper.find('p').text().trim() ).toBe( value );
    });
    
    test('Should not post the data onSubmit.', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} } );

        expect( setCategories ).not.toHaveBeenCalled();
    });

    test('Should call the setCategories and clean the input.', () => {
        const value = 'Hello Juan';

        wrapper.find('input').simulate('change', { target: { value } } );

        wrapper.find('form').simulate('submit', { preventDefault(){} } );

        expect( setCategories ).toHaveBeenCalled();

        expect( wrapper.find('input').prop('value') ).toBe( '' );
    });
    
});