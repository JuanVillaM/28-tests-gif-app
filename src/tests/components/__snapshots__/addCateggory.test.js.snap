// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <addCategory /> Should show himself correctly. 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
